runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/streaming/ActiveWorkState.java

```diff
@@ -124,11 +144,18 @@ private static String elapsedString(Instant start, Instant end) {
    */
   synchronized ActivateWorkResult activateWorkForKey(ExecutableWork executableWork) {
     ShardedKey shardedKey = executableWork.work().getShardedKey();
+    WorkIdWithShardingKey workIdWithShardingKey =
```

Comment:
could build this outside synchronized block

Reply:
done.